javascript/components/MapView.tsx

```diff
@@ -278,540 +257,603 @@ export interface MapViewState {
+         */
+        setCamera,
+        /**
+         * Takes snapshot of map with current tiles and returns a URI to the image
+         * @param  {Boolean} writeToDisk If true will create a temp file, otherwise it is in base64
+         * @return {String}
+         */
+        takeSnap,
+        /**
+         * Returns the current zoom of the map view.
+         *
+         * @example
+         * const zoom = await this._map.getZoom();
+         *
+         * @return {Number}
+         */
+        getZoom,
+        /**
+         * Returns the map's geographical centerpoint
+         *
+         * @example
+         * const center = await this._map.getCenter();
+         *
+         * @return {Array<Number>} Coordinates
+         */
+        getCenter,
+        /**
+         * Sets the visibility of all the layers referencing the specified `sourceLayerId` and/or `sourceId`
+         *
+         * @example
+         * await this._map.setSourceVisibility(false, 'composite', 'building')
+         *
+         * @param {boolean} visible - Visibility of the layers
+         * @param {String} sourceId - Identifier of the target source (e.g. 'composite')
+         * @param {String=} sourceLayerId - Identifier of the target source-layer (e.g. 'building')
+         */
+        setSourceVisibility,
+        /**
+         * Show the attribution and telemetry action sheet.
+         * If you implement a custom attribution button, you should add this action to the button.
+         */
+        showAttribution,
+        setNativeProps,
+      }));
+
+      const {_runNativeCommand, _runPendingNativeCommands, _onAndroidCallback} =
+        useNativeBridge(NATIVE_MODULE_NAME);
+      const logger = useRef<Logger>(Logger.sharedInstance());
+      // * start the logger before anyuseEffect
+      useOnce(() => {
+        logger.current.start();
+      });
+      const _nativeRef = useRef<RCTMGLMapViewRefType>();
+      const [isReady, setIsReady] = useState(false);
+
+      //? all this state values rerender the component, are they needed?
+      // * kept for the moment to avoid breaking changes
+      const [region, setRegion] = useState<RegionPayload | null>(null); // eslint-disable-line @typescript-eslint/no-unused-vars
+      const [width, setWidth] = useState(0); // eslint-disable-line @typescript-eslint/no-unused-vars
+      const [height, setHeight] = useState(0); // eslint-disable-line @typescript-eslint/no-unused-vars
+      const [isUserInteraction, setIsUserInteraction] = useState(false); // eslint-disable-line @typescript-eslint/no-unused-vars
+
+      useEffect(() => {
+        _setHandledMapChangedEvents(props);
+
+        return (): void => {
+          _onDebouncedRegionWillChange.clear();
+          _onDebouncedRegionDidChange.clear();
+          logger.current.stop();
+        };
+      }, []);
+
+      useEffect(() => {
+        _setHandledMapChangedEvents(props);
```

Comment:
This method will be called twice on the first mount, right? It will be called in the above hook and then here.

Maybe we shouldn't call it in the line 414?